src/main/java/seedu/address/model/note/Note.java

```diff
@@ -49,4 +62,7 @@ public String toString() {
     public int hashCode() {
         return Objects.hash(noteContents);
     }
```

Comment:
Since the equality checking was updated to include hasSameTags, the hashCode should be updated to ensure the tags are hashed as well, to maintain the contract between the equals and hashCode method.